doctorapp/src/components/LoginForm.vue

```diff
@@ -0,0 +1,129 @@
+<template>
+  <div class="form-container">
+    <h1>DoctorApp</h1>
+    <form @submit.prevent="login">
+      <div class="input-group">
+        <label for="solid-pod-url">Solid Pod URL:</label>
+        <input type="url" id="solid-pod-url" v-model="solidPodUrl" required>
+        <p v-if="showWarning" class="warning">Invalid Solid Pod URL. Please check and try again.</p>
+      </div>
+      <div class="btn-group">
+        <button @click.prevent="setDefaultIdp" :disabled="isLoading">No Pod?</button>
+        <button type="submit" :disabled="isLoading">
+          <span v-if="isLoading">Loading...</span>
+          <span v-else>Login</span>
+        </button>
+      </div>
+    </form>
+  </div>
+</template>
+
+<script lang="ts">
+import { ref } from 'vue';
+import {store} from '@/store';
+
+export default {
+  setup(props, { emit }) {
+    const solidPodUrl = ref<string>('');
+    const idp = ref<string>("https://css12.onto-deside.ilabt.imec.be/");
+    const showWarning = ref<boolean>(false);
+    const isLoading = ref<boolean>(false);
+
+    const login = async () => {
+      isLoading.value = true;
+      try {
+        const issuer = solidPodUrl.value.trim() !== '' ? solidPodUrl.value.trim() : idp.value;
+
+        // Attempt login
+        await store.session.login({
+          oidcIssuer: issuer,
+          redirectUrl: new URL('/home', window.location.href).toString(),
+          clientName: 'DOCTORAPP'
+        });
```

Comment:
Instead of try, catch & finally I would do `store.session.login().catch().then()`